pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -365,3 +359,98 @@ func (cs *ConntrackConnectionStore) fillL7EventInfo(l7EventMap map[connection.Tu
 		}
 	}
 }
+
+func (cs *ConntrackConnectionStore) getZones() []uint16 {
+	var zones []uint16
+	zones = append(zones, 0)
+	if cs.v4Enabled {
+		if cs.connectUplinkToBridge {
+			zones = append(zones, uint16(openflow.IPCtZoneTypeRegMark.GetValue()<<12))
+		} else {
+			zones = append(zones, openflow.CtZone)
+		}
+	}
+	if cs.v6Enabled {
+		if cs.connectUplinkToBridge {
+			zones = append(zones, uint16(openflow.IPv6CtZoneTypeRegMark.GetValue()<<12))
+		} else {
+			zones = append(zones, openflow.CtZoneV6)
+		}
+	}
+	return zones
+}
+
+func NewZoneZeroCache() ZoneZeroCache {
+	return ZoneZeroCache{
+		cache: map[string]*connection.Connection{},
+	}
+}
+
+// A cache holding zone zero connections for correlating the zone zero and antrea flows that make up an external flow.
+type ZoneZeroCache struct {
+	cache map[string]*connection.Connection
+}
+
+// Given a conn, generate a key that is unique to this connection
+// but can also be derived for the matching antrea ct_zone record.
+func (c ZoneZeroCache) generateKey(conn *connection.Connection) string {
+	destinationAddress := conn.FlowKey.DestinationAddress.String()
+	replyDestinationPort := strconv.FormatUint(uint64(conn.ProxySnatPort), 10)
+	return fmt.Sprintf("%s-%s", destinationAddress, replyDestinationPort)
```

Comment:
```suggestion
	destinationAddress := conn.FlowKey.DestinationAddress.String()
	return fmt.Sprintf("%s-%d", destinationAddress, conn.ProxySnatPort)
```

There's no need to convert it. Similarly in other places